provider/lxd/server.go

```diff
@@ -253,9 +273,48 @@ func (s *serverFactory) bootstrapLocalServer(svr Server) (Server, string, error)
 			hostAddress, connInfoAddresses,
 		)
 	}
+
+	// If the server is not a simple simple stream server, don't check the
+	// API version, but do report for other scenarios
+	if connInfoProtocol != string(lxd.SimpleStreamsProtocol) {
```

Comment:
I know this was in the old LXD client logic, but I can't see it would even come about.
The we never connect to a remote for wrapping in `Server` without implicitly using the LXD protocol.

Furthermore, in _container/lxd/connection.go_, we might even be able to get rid of the conditional in `connectImageRemote` and remove `Protocol` from `ServerSpec` altogether.

Reply:
I'd love to get rid of this check 😀 it feels wrong here.